src/util.c

```diff
@@ -872,6 +872,26 @@ int ld2string(char *buf, size_t len, long double value, ld2string_mode mode) {
     return 0;
 }
 
+/* Parses a version string on the form "major.minor.patch" and returns an
+ * integer on the form 0xMMmmpp. Returns -1 on parse error. */
+int version2num(const char *version) {
+    int v = 0, part = 0;
+    const char *p = version;
+    do {
+        if (*p >= '0' && *p <= '9') {
+            part = part * 10 + (unsigned)(*p - '0');
+        } else if (*p == '.') {
+            v = (v << 8) | part;
+            part = 0;
+        } else {
+            return -1;
+        }
+        p++;
+    } while (*p);
+    v = (v << 8) | part;
+    return v;
+}
+
```

Comment:
I think we could tighten up the validation a bit so the DOS attach risk is addressed. 

1. the string length should be bounded, i.e., 12 chars at most
2. part should not exceed 0xff
3. there should be exactly 2 "."

Reply:
I can't see how this is sensitive to any DOS attack. Invalid version numbers would just result in a version that's nonsense. The version parsing stops at the null terminator anyway. Do you think this computation is heavy for a long string? I think the network is the bottleneck in that case.

I can add the checks for other reasons though, like making it more strict to catch misbehaving clients. I think we shall document commands like REPLCONF better in the future to allow other programs/systems to connect as replicas.